test/boost/s3_test.cc

```diff
@@ -581,3 +581,50 @@ void client_access_missing_object(const client_maker_function& client_maker) {
 SEASTAR_THREAD_TEST_CASE(test_client_access_missing_object_minio) {
     client_access_missing_object(make_minio_client);
 }
+
+SEASTAR_THREAD_TEST_CASE(test_object_reupload) {
+    const sstring name(fmt::format("/{}/testobject-{}", tests::getenv_safe("S3_BUCKET_FOR_TEST"), ::getpid()));
+
+    semaphore mem(16 << 20);
+    auto cln = make_minio_client(mem);
+    auto close_client = deferred_close(*cln);
+
+    for (auto i : {1, 2}) {
+        testlog.info("Put object {}, iteration {}", name, i);
+        temporary_buffer<char> data = sstring("1234567890").release();
+        cln->put_object(name, std::move(data)).get();
+
+        size_t sz = cln->get_object_size(name).get();
+        BOOST_REQUIRE_EQUAL(sz, 10);
+
+        s3::stats st = cln->get_object_stats(name).get();
+        BOOST_REQUIRE_EQUAL(st.size, 10);
+    }
+
+    for (auto jumbo : {true, false}) {
+        for (auto i : {1, 2}) {
+            testlog.info("Upload object {}, iteration {} (with copy = {})", name, i, jumbo);
+            auto out = output_stream<char>(
+                // Make it 3 parts per piece, so that 128Mb buffer below
+                // would be split into several 15Mb pieces
+                jumbo ? cln->make_upload_jumbo_sink(name, 3) : cln->make_upload_sink(name));
+
+            static constexpr unsigned chunk_size = 1000;
```

Comment:
please either add a comment about what is being done here or use variables instead of numbers.